src/aligners/diagonal_transition.rs

```diff
@@ -115,6 +116,145 @@
 }
 
 impl<const N: usize> DiagonalTransition<AffineCost<N>> {
+    pub(super) fn track_path(
+        &self,
+        fronts: &mut Vec<Front<N>>,
+        a: &Sequence,
+        b: &Sequence,
+        mut s: usize,
+    ) -> (PATH, Cigar) {
+        let mut d = a.len() as i32 - b.len() as i32; //diagonal
+        let mut path: PATH = vec![];
+        let mut cigar = Cigar::default();
+
+        // The current position and affine layer.
+        let i = a.len();
+        let j = b.len();
+        // None for main layer.
+        let mut layer: Option<usize> = None;
+
+        path.push((i, j));
+
+        let mut save = |x: usize, y: usize| {
+            if let Some(last) = path.last() {
+                if *last == (x, y) {
+                    return;
+                }
+            }
+            path.push((x, y));
+        };
+
+        // let get_front = |cost, s1| &fronts[fronts.len() - cost as usize - (s0 - s1)];
+
+        'path_loop: while s > 0 || d != 0 || layer.is_some() {
+            let mut new_d = d; //new diagonal
+            let mut f = Fr::MIN;
+            let mut new_cost = s;
+            let mut flag = 0; //This flags shows do we need to save previous vertex
+            match self.gap_variant {
+                GapOpen => {
+                    // If None - we are on the main layer; if layer == Some (idx), then idx - index of the affine layer we are currently are
+                    if let Some(layer_idx) = layer {
+                        let command;
+                        match &self.cm.affine[layer_idx].affine_type {
+                            // I might have mixed up deletion and insertion again :(
+                            InsertLayer => {
+                                new_d = d + 1;
+                                command = CigarOp::AffineInsertion(layer_idx);
+                            }
+                            DeleteLayer => {
+                                new_d = d - 1;
+                                command = CigarOp::AffineDeletion(layer_idx);
+                            }
+                            _ => todo!(),
+                        }
+                        if fronts[s - self.cm.affine[layer_idx].extend as usize]
+                            .affine_mut(layer_idx)[new_d]
```

Comment:
Replace this by some condition on furthest reaching point in diagonal new_d vs
the current f.r. point